cheatsheets/DotNet_Security_Cheat_Sheet.md

```diff
@@ -246,7 +248,226 @@ EXEC strQry // SQL Injection vulnerability!
 
 DO: Practise Least Privilege - Connect to the database using an account with a minimum set of permissions required to do it's job i.e. not the sa account
 
-## A2 Weak Account management
+### [OS Injection](OS_Command_Injection_Defense_Cheat_Sheet.md#net)
+
+DO: Use [System.Diagnostics.Process.Start](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.start?view=netframework-4.7.2) to call underlying OS functions.
+
+e.g
+
+``` csharp
+System.Diagnostics.Process process = new System.Diagnostics.Process();
+System.Diagnostics.ProcessStartInfo startInfo = new System.Diagnostics.ProcessStartInfo();
+startInfo.FileName = "validatedCommand";
+startInfo.Arguments = "validatedArg1 validatedArg2 validatedArg3";
+process.StartInfo = startInfo;
+process.Start();
+```
+
+DO: Use whitelist validation on all user supplied input. Input validation prevents improperly formed data from entering an information system. For more information please see the [Input Validation Cheat Sheet](Input_Validation_Cheat_Sheet.md).
+
+e.g Validating user input using [IPAddress.Parse Method](https://docs.microsoft.com/en-us/dotnet/api/system.net.ipaddress.parse?view=netframework-4.8)
+
+``` csharp
+//User input
+string ipaddress = "127.0.0.1";
+ 
+//check to make sure an ip address was provided    
+if (string.IsNullOrEmpty(address))  
+{   
+	try
+	{
+		// Create an instance of IPAddress for the specified address string (in 
+		// dotted-quad, or colon-hexadecimal notation).
+		IPAddress address = IPAddress.Parse(ipAddress);
+
+		// Display the address in standard notation.
+		return address.ToString();
+	}
+	catch(ArgumentNullException e)
+	{
+		//ipaddress is not of type IPaddress
+		...
+	}
+    ...
+}
+ ```
+ 
+### LDAP injection
+
+Almost any characters can be used in Distinguished Names. However, some must be escaped with the backslash "\" escape character. Active Directory requires that the following characters be escaped:
+
+| Character | Escape character |
+|-----------|:-----:|
+|Comma	|,|
+|Backslash character|\|
+|Pound sign (hash sign)|#|
+|Plus sign|+|
+|Less than symbol|<|
+|Greater than symbol|>|
+|Semicolon|;|
+|Double quote (quotation mark)|"|
+|Equal sign|=|
+|Leading or trailing spaces| | 
+
+NB: The space character must be escaped only if it is the leading or trailing character in a component name, such as a Common Name. Embedded spaces should not be escaped.
+
```

Comment:
At the end of every section add a link like this (you have added it on some section and I think that it will be great to apply it on every section):

*More information can be found here: XXX Cheat Sheet* where `XXX` is the name and relative link to the dedicated cheat sheet about the topic of the section (if they are a sub section about .NET in the CS then point to it directly).